### DISCUSSION TITLE
Uploader Review

### DISCUSSION THREAD

**Discussion by richard1015** [jdf2e/nutui-react#10] [General]
- 功能、基本交互测试
   - [ ] 缺少手动上传模式
   - [ ] 需要同步更新一下 nutui vue 的 `upload.ts` 文件 (期间有修复相关bug)
- 文档完整性验证
   - [ ] import { Uploader } from '@nutui/nutui-react';
   - [ ] ...
- props 所有属性测试
   - [ ] ...
   - [ ] ...
- props 异步修改组件是否动态更新
   - [ ] ...
   - [ ] ...
- emits 所有事件测试
   - [ ] ...
   - [ ] ...
- npm run build 子组件是否被构建
   - [ ] ...
   - [ ] ...
- 其它
   - [ ] ...
   - [ ] ...

**Comment by junjun666** [comment]
已新增手动上传模式的开发
文档部分在其他分支上已更新，待合并

**Comment by xiaoyatong** [comment]
1. removeImage 是什么意思？
2. 上传图片的预览的缩放最好等比例缩放
![image](https://user-images.githubusercontent.com/84436086/139427599-bb177c63-153a-4fca-a7e6-4c2a70cd0b15.png)

3. 点击上传后，建议过滤文件类型。
![image](https://user-images.githubusercontent.com/84436086/139427805-14a72ae6-0350-4330-b2d0-4cfd508c5195.png)
4. 上限没有限制住，复现：先上传1个，再上传5个。
![image](https://user-images.githubusercontent.com/84436086/139428239-3f5ad632-78b6-4aec-800f-5cdbe3dbc960.png)

5. 点击上传后，不知道在哪儿展示上传。demo完善一下~
![image](https://user-images.githubusercontent.com/84436086/139428357-12928a6a-fbd7-48cd-97e2-f12e52bd6ea4.png)

**Comment by junjun666** [comment]
@xiaoyatong 
1.removeImage 是文件删除之前的状态,doc文档中有所标注。和vue的delete方法对应，因delete' cannot be called on an identifier in strict mode。所以这里采用removeImage名字来替代。
2.对于长图压缩的问题已调整
3.目前vue的源码版本type=“file”是支持上传文件类型的，在源码中读取文件中会有一个判断，如果是图像会展示图像照片，如果不是image类型则不展示图像。所以这里是否过滤文件类型，还有待商榷。
4.上限没有限制住的问题已经修复，vue的版本也存在类似问题，请关注@richard1015 
5.这里的上传文件是一个自定义的上传样式，所以这个demo不会像其他组件展示出来，而是根据配置的的url（服务器地址）来直接上传。
